engine/src/main/java/org/hibernate/validator/spi/nodenameprovider/Property.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.spi.nodenameprovider;
+
+public interface Property {
+	Class<?> get();
```

Comment:
It's unclear what's the purpose of this method. We should have javadoc for all public classes and their methods. Please also note that we should have an hierarchy of property interfaces as mentioned before.
```java
interface Property {
	String getName();
}

interface JavaBeanProperty extends Property {
	Annotation[] getDeclaredAnnotations();
	Class<?> getDeclaringClass();
}
```
It is important that we don't expose any methods in property interface that wouldn't make sense for non JavaBean properties.